spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitMessagingTemplate.java

```diff
@@ -195,7 +197,13 @@ public void convertAndSend(String exchange, String routingKey, Object payload,
 	@Override
 	protected void doSend(String destination, Message<?> message) {
 		try {
-			this.rabbitTemplate.send(destination, createMessage(message));
+			if (message.getHeaders().get(AmqpHeaders.PUBLISH_CONFIRM_CORRELATION) instanceof CorrelationData) {
```

Comment:
May we perform this `get()` only once?

Reply:
I was trying to avoid a cast, but I suppose it's better to do that.